core/src/test/java/com/linecorp/armeria/common/stream/StreamMessageInputStreamTest.java

```diff
@@ -318,4 +323,112 @@ void toInputStream_executor() throws IOException {
         assertThat(actual).isEqualTo(expected);
         assertThat(inputStream.available()).isZero();
     }
+
+    @Test
+    void httpDataConverter_empty() throws IOException {
+        final StreamMessage<Integer> streamMessage = StreamMessage.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+        final InputStream inputStream = streamMessage
+                .toInputStream(x -> {
+                    if (x % 2 == 0) {
+                        return HttpData.empty();
+                    }
+                    return HttpData.wrap(x.toString().getBytes());
+                });
+
+        final byte[] expected = ImmutableList.of(1, 3, 5, 7, 9)
+                                             .stream()
+                                             .map(x -> x.toString().getBytes())
+                                             .reduce(Bytes::concat).get();
+
+        final ByteBuf result = Unpooled.buffer();
+        int read;
+        while ((read = inputStream.read()) != -1) {
+            result.writeByte(read);
+        }
+
+        final int readableBytes = result.readableBytes();
+        final byte[] actual = new byte[readableBytes];
+        for (int i = 0; i < readableBytes; i++) {
+            actual[i] = result.readByte();
+        }
+        result.release();
+        assertThat(actual).isEqualTo(expected);
+        assertThat(inputStream.available()).isZero();
+    }
+
+    @Test
+    void httpDataConverter_error_thrown() throws IOException {
+        final StreamMessage<Integer> streamMessage = StreamMessage.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+        final InputStream inputStream = streamMessage
+                .toInputStream(x -> {
+                    if (x == 6) {
+                        return Exceptions.throwUnsafely(new RuntimeException());
```

Comment:
We can throw a non-checked exception in a `Function<T, R>`.
```suggestion
                        throw new RuntimeException();
```